safewalk2-platform/azuredeploy.json

```diff
@@ -0,0 +1,900 @@
+            "type": "Microsoft.Compute/virtualMachines/extensions",
+            "name": "[concat(parameters('vmPrefix'), '-server-2/installcustomscript')]",
+            "apiVersion": "2017-03-30",
+            "location": "[variables('location')]",
+            "dependsOn": [
+                "[concat('Microsoft.Compute/virtualMachines/', parameters('vmPrefix'), '-server-2')]",
+                "[concat('Microsoft.Compute/virtualMachines/', parameters('vmPrefix'), '-gateway')]",
+                "[resourceId('Microsoft.Compute/virtualMachines/extensions', concat(parameters('vmPrefix'), '-server-1'), 'installcustomscript')]"
+            ],
+            "properties": {
+                "publisher": "Microsoft.Azure.Extensions",
+                "type": "CustomScript",
+                "typeHandlerVersion": "2.0",
+                "autoUpgradeMinorVersion": true,
+                "settings": {
+                    "fileUris": [
+                        "[concat(parameters('_artifactsLocation'), '/', variables('ScriptFolder'), '/', 'configure_safewalk_custom_script_slave.sh', parameters('_artifactsLocationSasToken'))]",
+                        "[concat(parameters('_artifactsLocation'), '/', variables('ScriptFolder'), '/', 'set_root_password.sh', parameters('_artifactsLocationSasToken'))]",
+                        "[concat(parameters('_artifactsLocation'), '/', variables('ScriptFolder'), '/', 'safewalk_create_gateway.sh', parameters('_artifactsLocationSasToken'))]",
+                        "[concat(parameters('_artifactsLocation'), '/', variables('ScriptFolder'), '/', 'safewalk_bdr_join.sh', parameters('_artifactsLocationSasToken'))]",
+                        "[concat(parameters('_artifactsLocation'), '/', variables('ScriptFolder'), '/', 'safewalk_make_partitions.sh', parameters('_artifactsLocationSasToken'))]",
+                        "[concat(parameters('_artifactsLocation'), '/', variables('ScriptFolder'), '/', 'safewalk_renew_secrets.sh', parameters('_artifactsLocationSasToken'))]",
+                        "[concat(parameters('_artifactsLocation'), '/', variables('ScriptFolder'), '/', 'safewalk_upgrade.sh', parameters('_artifactsLocationSasToken'))]",
+                        "[concat(parameters('_artifactsLocation'), '/', variables('ScriptFolder'), '/', 'safewalk_iptables.sh', parameters('_artifactsLocationSasToken'))]",
+                        "[concat(parameters('_artifactsLocation'), '/', variables('ScriptFolder'), '/', 'setup_snmp.sh', parameters('_artifactsLocationSasToken'))]"
+                    ],
+                    "commandToExecute": "[concat('sh configure_safewalk_custom_script_slave.sh ', parameters('SafewalkServerRootPassword'), ' ', parameters('SafewalkGatewayIP'), ' ', reference(concat('Microsoft.Network/publicIPAddresses/', variables('gatewayPublicIPName')), '2017-04-01').ipAddress, ' ', parameters('SafewalkGatewayRootPassword'), ' ', parameters('FirstSafewalkServerIP'), ' ', parameters('SecondSafewalkServerIP'))]"

```

Comment:
put this in protectedSettings since it contains secrets

Reply:
fileUris is not compatible with protectedSettings. How I need to do to convine fileUris with protectedSettings since I need the scripts to be executed in the VMs?

Thansk!

Reply:
Forget it.. Fixed.